src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -163,15 +163,22 @@ public void EndConnect(IAsyncResult asyncResult)
                 NetEventSource.Enter(NetEventSource.ComponentType.Socket, this, nameof(EndConnect), asyncResult);
             }
 
-            Socket s = Client;
-            if (s == null)
+            if (TaskToApm.IsTask(asyncResult))
```

Comment:
This assumes that `Socket.BeginConnect` doesn't return a Task-based `IAsyncResult.`  Could create some pain later, I guess.  I wonder if it would be worth moving the remaining `TcpClient.Connect` overload to also call the equivalent `ConnectAsync` method?